src/lightning_app/core/constants.py

```diff
@@ -77,8 +77,32 @@ def enable_multiple_works_in_default_container() -> bool:
 
 
 def get_cloud_queue_type() -> Optional[str]:
-    return os.getenv("LIGHTNING_CLOUD_QUEUE_TYPE", None)
+    value = os.getenv("LIGHTNING_CLOUD_QUEUE_TYPE", None)
+    if value is None and enable_preemptible_works():
+        value = "http"
+    return value
 
 
 # Number of seconds to wait between filesystem checks when waiting for files in remote storage
 REMOTE_STORAGE_WAIT = 0.5
+
+
+# preemptible support
+def enable_preemptible_works() -> bool:
+    return bool(int(os.getenv("LIGHTNING_ENABLE_PREEMPTIBLE_WORKS", "0")))
```

Comment:
Is this a variable that the user sets, or something set by the cloud? Could you maybe add the answer as a comment here?

Reply:
This is currently an experimental feature, so on the user.